client/client.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+)
+
+// HTTPClient provides a properly configured http.Client object used to send
+// requests out to a ContainerPilot process's control socket.
+type HTTPClient struct {
+	http.Client
+	socketPath string
+}
+
+var socketType = "unix"
+
+func socketDialer(socketPath string) func(string, string) (net.Conn, error) {
+	return func(_, _ string) (net.Conn, error) {
+		return net.Dial(socketType, socketPath)
+	}
+}
+
+// NewHTTPClient initializes an client.HTTPClient object by configuring it's
+// socketPath for HTTP communication through the local file system.
+func NewHTTPClient(socketPath string) (*HTTPClient, error) {
+	if socketPath == "" {
+		err := errors.New("control server not loading due to missing config")
+		return nil, err
+	}
+
+	client := &HTTPClient{}
+	client.Transport = &http.Transport{
+		Dial: socketDialer(socketPath),
+	}
+
+	return client, nil
+}
+
+// Reload makes a request to the reload endpoint of a ContainerPilot process.
+func (c HTTPClient) Reload() (*http.Response, error) {
+	resp, err := c.Post("http://control/v3/reload", "application/json", nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		mesg := fmt.Sprintf("expected 200 but got %v\n%+v", resp.StatusCode, resp)
```

Comment:
There was a point in which it made sense to broaden the usability of the client package itself. Looking at it a bit further now, it makes more sense to only return the status code and handle the error message closer to where the client is being implemented. That'll allow the implementer (`subcommands`) to make a decision of how to handle the response.

I feel like the overall return of the response isn't warranted right now either. We're not making use of it and can revisit if we need it in the future.

Also, I had questioned the closing of `resp.Body` but wasn't familiar enough to pick up on the repercussions. Probably the old Rubyist in me.... 🛩